services/brig/src/Brig/API/Util.hs

```diff
@@ -36,13 +37,21 @@ lookupProfilesMaybeFilterSameTeamOnly self us = do
     Just team -> filter (\x -> profileTeam x == Just team) us
     Nothing -> us
 
+--------------------------------------------------------------------------------
+-- Federation
+
+viewFederationDomain :: MonadReader Env m => m (Maybe Domain)
+viewFederationDomain = view (settings . enableFederationWithDomain)
+
+isFederationEnabled :: MonadReader Env m => m Bool
+isFederationEnabled = isJust <$> viewFederationDomain
+
 -- FUTUREWORK(federation, #1178): implement function to resolve IDs in batch
 
 -- | this exists as a shim to find and mark places where we need to handle 'OpaqueUserId's.
 resolveOpaqueUserId :: MonadReader Env m => OpaqueUserId -> m (MappedOrLocalId Id.U)
 resolveOpaqueUserId (Id opaque) = do
-  mEnabled <- view (settings . enableFederation)
-  case fromMaybe defEnableFederation mEnabled of
+  isFederationEnabled >>= \case
     False ->
       -- don't check the ID mapping, just assume it's local
```

Comment:
How will this work if we switch an unfederated backend with users to a federated one?  I suppose there cannot be any remote users.

What about switching a federated backend into a non-federated one?  Then we had remote users, but now they are assumed to be local, right?  Should we just clearly state that turning the switch this way is not allowed?  Or is there a reason why it's not a problem?

Reply:
Enabling federation is completely fine, turning it off is more complicated. Probably we should make it clear that this is not recommended/supported for now.

At the moment the effect is that the remote users and conversations would just seem to have disappeared. Handling this more elegantly is probably a lot of work for both backend and clients.